code/modules/mob/living/carbon/human/species_types/diona.dm

```diff
@@ -0,0 +1,383 @@
+	staminamod = 0.7
+	meat = /obj/item/food/meat/slab/human/mutant/diona
+	exotic_blood = /datum/reagent/consumable/chlorophyll
+	species_gibs = null //Someone please make this like, xeno gibs or something in the future. I cant be bothered to fuck around with gib code right now.
+	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | ERT_SPAWN | RACE_SWAP
+	species_language_holder = /datum/language_holder/diona
+	bodytemp_normal = (BODYTEMP_NORMAL - 22) // Body temperature for dionae is much lower then humans as they are plants, supposed to be 15 celsius
+	speedmod = 1.2 // Dionae are slow.
+	species_height = SPECIES_HEIGHTS(0, -1, -2) //Naturally tall.
+	swimming_component = /datum/component/swimming/diona
+	inert_mutation = DRONE
+	deathsound = "sound/emotes/diona/death.ogg"
+	species_bitflags = NOT_TRANSMORPHIC
+
+	mutanteyes = /obj/item/organ/eyes/diona //SS14 sprite
+	mutanttongue = /obj/item/organ/tongue/diona //Dungeon's sprite
+	mutant_brain = /obj/item/organ/brain/diona //SS14 sprite
+	mutantliver = /obj/item/organ/liver/diona //Dungeon's sprite
+	mutantlungs = /obj/item/organ/lungs/diona //Dungeon's sprite
+	mutantstomach = /obj/item/organ/stomach/diona //SS14 sprite
+	mutantears = /obj/item/organ/ears/diona //SS14 sprite
+	mutant_heart = /obj/item/organ/heart/diona //Dungeon's sprite
+
+	species_chest = /obj/item/bodypart/chest/diona
+	species_head = /obj/item/bodypart/head/diona
+	species_l_arm = /obj/item/bodypart/l_arm/diona
+	species_r_arm = /obj/item/bodypart/r_arm/diona
+	species_l_leg = /obj/item/bodypart/l_leg/diona
+	species_r_leg = /obj/item/bodypart/r_leg/diona
+
+	var/datum/action/diona/split/split_ability //All dionae start with this, this is for splitting apart completely.
+	var/datum/action/diona/partition/partition_ability //All dionae start with this as well, this is for splitting off a nymph from food.
+	var/mob/living/simple_animal/hostile/retaliate/nymph/drone = null
+
+	var/time_spent_in_light
+	var/informed_nymph = FALSE //If the user was informed that they can release a nymph via food.
+
+/datum/species/diona/spec_life(mob/living/carbon/human/H)
+	if(H.fire_stacks < 1)
+		H.adjust_fire_stacks(1) //VERY flammable
+	if(H.nutrition < NUTRITION_LEVEL_STARVING)
+		H.take_overall_damage(1,0)
+	if(H.stat != CONSCIOUS)
+		H.remove_status_effect(STATUS_EFFECT_PLANTHEALING)
+	if((H.health <= H.crit_threshold)) //Shit, we're dying! Scatter!
+		split_ability.Trigger(TRUE)
+	if(H.nutrition > NUTRITION_LEVEL_WELL_FED && !informed_nymph)
+		informed_nymph = TRUE
+		to_chat(H, "<span class='warning'>You feel sufficiently satiated to allow a nymph to split off from your gestalt!")
+	if(partition_ability)
+		partition_ability.UpdateButtonIcon()
+	if(H.nutrition > NUTRITION_LEVEL_ALMOST_FULL)
+		H.set_nutrition(NUTRITION_LEVEL_ALMOST_FULL)
+	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
+	if(isturf(H.loc)) //else, there's considered to be no light
```

Comment:
return early

Reply:
I am unsure what you mean by this.

Reply:
reverse the condition and return early

Reply:
there's another check so you cant do that anyway